Orange/widgets/visualize/owsieve.py

```diff
@@ -29,11 +29,29 @@ class ChiSqStats:
     pair of attributes. The class is also used for ranking.
     """
     def __init__(self, data, attr1, attr2):
-        self.observed = get_contingency(data, attr1, attr2)
-        self.n = np.sum(self.observed)
-        self.probs_x = self.observed.sum(axis=0) / self.n
-        self.probs_y = self.observed.sum(axis=1) / self.n
-        self.expected = np.outer(self.probs_y, self.probs_x) * self.n
+        attr1 = data.domain[attr1]
+        attr2 = data.domain[attr2]
+        if attr1.is_discrete and not attr1.values or \
+                        attr2.is_discrete and not data.domain[attr2].values:
```

Comment:
Unlike `attr1`, `attr2` is again mapped through `data.domain`.
Also indentation?